odl/discr/discr_mappings.py

```diff
@@ -501,47 +401,39 @@ def __init__(self, ip_fspace, grid, dspace, order='C'):
             means the first grid axis varies fastest, the last most
             slowly, 'F' vice versa.
         """
-        FunctionSetMapping.__init__(self, 'extension', ip_fspace,
-                                    grid, dspace, order, linear=True)
+        if not isinstance(ip_fspace, FunctionSpace):
+            raise TypeError('function space {!r} is not a `FunctionSpace` '
+                            'instance.'.format(ip_fspace))
+        if not isinstance(ip_fspace.domain, IntervalProd):
+            raise TypeError('function space domain {!r} is not an '
+                            '`IntervalProd` instance.'.format(ip_fspace))
 
-    # TODO: Implement _apply()
+        FunctionSetMapping.__init__(self, 'extension', ip_fspace, grid, dspace,
+                                    order, linear=True)
 
-    def _call(self, inp):
-        """The raw call method for out-of-place evaluation.
+    def _call(self, x, out=None):
+        """The raw method for out-of-place evaluation.
 
         Parameters
         ----------
-        inp : `NtuplesVector`
+        x : `NtuplesVector`
             The array of numbers to be interpolated
 
         Returns
         -------
-        outp : `FunctionSetVector`
+        out : `FunctionSetVector`
             A function (nearest-neighbor) interpolating at a given
             point or array of points.
 
         Examples
         --------
-        Let's define the complex function space :math:`L^2` on a
-        rectangle:
-
         TODO: implement an example!
-
         """
-        def func(*x):
-            x = np.atleast_1d(x).squeeze()
-            values = interpnd(points=self.grid.coord_vectors,
-                              values=inp.data.reshape(self.grid.shape,
-                                                      order=self.order),
-                              method='nearest',
-                              xi=x,
-                              fill_value=None)  # Allow points outside
-            return values[0] if values.shape == (1,) else values
+        # TODO: implement
```

Comment:
It would only work for the "array" vectorized evaluation, not meshgrids. I'll rather make two interpolator classes as for the nearest, they should be very easy to generalize from the latter.